include/smack/SmackRep.h

```diff
@@ -150,6 +166,14 @@ class SmackRep {
   virtual const Stmt* alloca(llvm::AllocaInst& i);
   virtual const Stmt* memcpy(const llvm::MemCpyInst& msi);
   virtual const Stmt* memset(const llvm::MemSetInst& msi);
+  virtual const Stmt* loadAsBytes(const llvm::LoadInst& li);
+  virtual const Stmt* storeAsBytes(const llvm::StoreInst& si);
```

Comment:
As long as you know that `val` is the operand of an instruction, you could retrieve the instruction(s) that use it via the `llvm::Value::users()` method, then the `Function` containing it as above.

However, I'm not sure that we're on the same page here. My intent was to suggest that both `storeAsBytes` methods get merged into one `store(const llvm::Value* addr, const llvm::Value* val)` method. This should be possible, since the `region` and `size` arguments are anyhow computed from `addr`. Neither of the original `storeAsBytes` methods used the containing `Function` object; why is that needed now?